### PR TITLE
Fix regressions from #6

### LINKED ISSUE #26: gcc-4.4 and gcc-4.6 were broken by #6
#6 broke older compilers

This drove addition of older compilers into CI.

per discussion, Boost.Config claims these compilers support variadic templates but their support is incomplete, and we need to do something special here.  See discussion on #6.

Also see https://github.com/boostorg/assign/pull/6#issuecomment-429896586

### LINKED ISSUE #28: Add build logic to load each header independently to make sure it compiles
There is an example in the uuid test Jamfile one can follow.  This will ensure each header can be included independently and each includes everything it requires.  This is based on regressions in #6 that were not detected by the current build process.

### PR DESCRIPTION
**Pull request by jeking3**
This fixes #26 
This fixes #28

### CODE REVIEW

**Comment by jeking3** [comment]
That went well... :|

File: include/boost/assign/list_of.hpp [hunk 1]
```diff
@@ -418,7 +424,7 @@ namespace assign_detail
             return *this;
         }
 
-#if defined(BOOST_NO_CXX11_VARIADIC_TEMPLATES) || defined(BOOST_NO_CXX11_RVALUE_REFERENCES)
+#if BOOST_ASSIGN_USE_VARIADIC_TEMPLATES
```
[the remark this answers is not in the document]
Comment (jeking3) [reply include/boost/assign/list_of.hpp]: Thanks, I was in a rush.  That's why we have pull requests. :)  I'll get back to it shortly.

**Comment by jeking3** [comment]
... just waiting for the travis CI backlog to clear and run this build ...

**Comment by morinmorin** [comment]
Thanks, LGTM for the .hpp files.

**Pull request merged**